app/questionnaire/questionnaire_schema.py

```diff
@@ -202,132 +211,159 @@ def _section_ids_associated_to_list_name(self, list_name: str) -> List:
         return section_ids
 
     @staticmethod
-    def get_blocks_for_section(section):
+    def get_blocks_for_section(section: Mapping) -> Iterator[ImmutableDict]:
```

Comment:
Since we said we are making the return types as specific as possible, this should return a `Generator`?. Few other occurrences also.

Reply:
Good spot, this was done before we agreed on the return types and I hadn't gone back to it. I have just tried changing it and there is one downside, the return type for these when using `Generator` is `Generator[ImmutableDict, None, None]`. Thoughts?

For reference the my docs can be found at https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators

Reply:
I'm okay with that, just how it has to be but does make it specific.

Reply:
Ok, done.